homeassistant/components/sensor/zestimate.py

```diff
@@ -0,0 +1,144 @@
+"""
+Support for zestimate data from zillow.com.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.zestimate/
+"""
+import logging
+from datetime import timedelta
+
+import voluptuous as vol
+import requests
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (CONF_API_KEY,
+                                 CONF_NAME, STATE_UNKNOWN, ATTR_ATTRIBUTION)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+REQUIREMENTS = ['xmltodict==0.11.0']
+
+_LOGGER = logging.getLogger(__name__)
+_RESOURCE = 'http://www.zillow.com/webservice/GetZestimate.htm'
+
+CONF_ZPID = 'zpid'
+CONF_ATTRIBUTION = "Data provided by Zillow.com"
+
+DEFAULT_NAME = 'Zestimate'
+NAME = 'zestimate'
+ZESTIMATE = '{}:{}'.format(DEFAULT_NAME, NAME)
+
+ICON = 'mdi:home-variant'
+
+ATTR_AMOUNT = 'amount'
+ATTR_CHANGE = 'amount_change_30days'
+ATTR_CURRENCY = 'amount_currency'
+ATTR_LAST_UPDATED = 'amount_last_updated'
+ATTR_VAL_HI = 'valuation_range_high'
+ATTR_VAL_LOW = 'valuation_range_low'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_API_KEY): cv.string,
+    vol.Required(CONF_ZPID): vol.All(cv.ensure_list, [cv.string]),
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+})
+
+
+# Return cached results if last scan was less then this time ago.
+MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=30)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Zestimate sensor."""
+
+    name = config.get(CONF_NAME)
+    properties = config[CONF_ZPID]
+    params = {'zws-id': config[CONF_API_KEY]}
+
+    sensors = []
+    for zpid in properties:
+        params['zpid'] = zpid
+        sensors.append(ZestimateDataSensor(name, params))
+    add_devices(sensors, True)
+
+
+class ZestimateDataSensor(Entity):
+    """Implementation of a Zestimate sensor."""
+
+    def __init__(self, name, params):
+        """Initialize the sensor."""
+        self._name = name
+        self.params = params
+        self.data = None
+        self.address = None
+        self._state = None
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        try:
+            return round(float(self._state), 1)
+        except ValueError:
+            return STATE_UNKNOWN
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        attributes = {}
+        if self.data is not None:
+            attributes['Amount'] = self.data[ATTR_AMOUNT]
```

Comment:
Attributes should be lowercase snakecase.

Reply:
It seems `self.data` is a subset of attributes, so just update attributes with `self.data` and then set address and attribution.